server/src/test/java/org/opensearch/index/store/RemoteSegmentStoreDirectoryTests.java

```diff
@@ -988,6 +988,53 @@ public void testDeleteStaleCommitsActualDelete() throws Exception {
         verify(remoteMetadataDirectory).deleteFile(metadataFilename3);
     }
 
+    public void testDeleteStaleCommitsActualDeleteWithLocks() throws Exception {
```

Comment:
nit - should we add comments to suggest what the tests are doing? Going through the code definitely helps understand it, but it can be easier otherwise.

Reply:
Test block has the comments in it.